proposals/22-04-localize-command.md

```diff
@@ -0,0 +1,331 @@
+    │                               │   └── pod.yaml
+    │                               ├── dev
+    │                               ├── kustomization.yaml
+    │                               ├── production
+    │                               └── staging
+    └── overlay
+        ├── kustomization.yaml
+        └── localized-files
+            └── github.com
+                └── kubernetes-sigs
+                    └── kustomize
+                        └── v1.0.6
+                            └── examples
+                                └── helloWorld
+                                    └── deployment.yaml
+```
+```shell
+# localized-overlay/overlay/kustomization.yaml
+resources:
+  - ../base
+  - ./localized-files/github.com/kubernetes-sigs/kustomize/examples/helloWorld/deployment.yaml
+```
+```shell
+# localized-overlay/base/kustomization.yaml
+resources:
+  - ./localized-files/github.com/kubernetes-sigs/kustomize/examples/multibases
+  - ./localized-files/github.com/kubernetes-sigs/kustomize/examples/helloWorld/configMap.yaml
+```
+
+Now, I upload `localized-overlay` from my local setup to my company’s internal package management site. I change the
+commands in my CI/CD pipeline to pull `localized-overlay` before running `kustomize build localized-overlay`, and the
+command executes successfully!
+
+### Risks and Mitigations
+
+One could argue that while uploading the localized `newDir` to a repository, a user could accidentally leak Secrets that
+were originally remote in a more private repo. This event is not very likely, as the servers that the user intends to
+consume the localized kustomizations often only have access to internal, private networks and repos. Nonetheless, users
+should have enough context in the case of Secrets to make the right decision. Users should have a basic understanding of
+the files that their `target` kustomizations reference and of the files that they plan to upload to repos. Secret
+configurations are also not too difficult to identify.
+
+Exec binaries that KRM functions reference are a different story. `kustomize localize` downloads remote exec binaries
+that, if malicious, are capable of almost anything during subsequent `kustomize build` calls. The command mitigates this
+risk by leaving these downloaded exec binaries without executable permissions and warning the user, as mentioned in
+**Proposal**. `kustomize build` can only run the exec binary after the user deems the binary safe and changes its
+permissions.
+
+Still another risk may be that if a user's kustomization tree is large, `kustomize localize` may be copying files from
+unexpected locations. The command mitigates this risk with the `scope` flag. If not set, `kustomize localize` only
+copies files in `target`. Otherwise, the user specifies `scope`and understands that `kustomize localize` only copies
+files in `scope`. The qualification that the command can only localize `target`s that follow load restrictions helps
```

Comment:
Nit: they understand it because it will be an explicit error, right?

Reply:
Yes, reworded this section.